File: pyround1/1436.旅行终点站.py
#
# @lc app=leetcode.cn id=1436 lang=python3
#
# [1436] 旅行终点站
#

# @lc code=start
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        P0 = [path[0] for path in paths]
        for path in paths:
            if path[1] not in P0:
                return path[1]
# @lc code=end

File: 1436.旅行终点站.cpp
/*
 * @lc app=leetcode.cn id=1436 lang=cpp
 *
 * [1436] 旅行终点站
 */

// @lc code=start
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_map<string, int>res;
        for(int i = 0; i < paths.size(); i++){
            res[paths[i][0]] += 1;
            
        }
        for(int i = 0; i < paths.size(); i++){
            res[paths[i][1]] -= 1;
            if(res[paths[i][1]] < 0)
                return paths[i][1];
        }
        return "";
    }
};
// @lc code=end

